### DISCUSSION TITLE
Add new category/tag from the admin sidebar

### DISCUSSION THREAD

**Discussion by jakehopking** [payloadcms/payload#1096] [Q&A] [answered]
Hi - I've created a Category collection, see the code below. 

Is it possible to create new categories from outside of the Categories collection?

I'd like to be able to assign and create new categories from within the Media collection (I'm planning on adding a tags collection next, and this use case will be even more important).

Currently I can obviously assign a category to an image, but as you can see from the screenshot, I can't create new ones:
![Screenshot 2022-09-08 at 16 18 26](https://user-images.githubusercontent.com/3432643/189162851-4439f57f-9d78-45d8-a430-e966cd8048ca.png)
☝️ What am I missing here?

Category collection:
```typescript
import { CollectionConfig } from 'payload/types';
import slug from '../fields/slug';

export type CategoryType = {
  title: string;
  slug: string;
};

const Category: CollectionConfig = {
  slug: 'categories',
  admin: {
    useAsTitle: 'title',
  },
  access: {
    read: () => true,
  },
  fields: [
    {
      name: 'title',
      label: 'Title',
      type: 'text',
      required: true,
    },
    slug(),
  ],
};

export { Category };
```
Categories field:
```typescript
import { Field } from 'payload/types';

export const categories: Field = {
  name: 'categories',
  label: 'Categories',
  type: 'relationship',
  relationTo: 'categories',
  hasMany: true,
  admin: {
    position: 'sidebar',
  },
};
```
Then adding to my Media photos collection:
```typescript
import { CollectionConfig } from 'payload/types';
import { categories } from '../fields/categories';
import { ImageSizeNames, ImageSizes, MediaSlugs } from '../tokens';

const MediaPhotos: CollectionConfig = {
  slug: MediaSlugs.mediaPhotos,
  access: {
    read: (): boolean => true, // Everyone can read Media
  },
  admin: {
    useAsTitle: 'filename',
  },
  upload: {
    adminThumbnail: ImageSizeNames.thumbnail,
    imageSizes: [...ImageSizes.PhotoImageSizes],
    staticURL: '/media/photos',
    staticDir: 'media/photos',
  },
  fields: [
    {
      name: 'alt',
      label: 'Alt Text',
      type: 'text',
      required: true,
    },
    categories, // <--- Here
  ],
};

export { MediaPhotos };
```

Thank you ❤️

**Comment by jmikrut** [comment, marked as answer]
We are building a way to do this as our next big feature! Great question!

**Reply by jakehopking** [reply]
Thanks for the speedy reply. Glad I asked... else could've ended up going down a bit of a rabbit hole I imagine.

Do you have a rough ETA on that feature hitting production?

**Reply by jmikrut** [reply]
Within 30 days 😎

**Reply by jmikrut** [reply]
Hey @jakehopking this is now built!

https://github.com/payloadcms/payload/pull/1230

Not merged yet. Gotta write some tests but just a quick heads-up!

**Reply by jakehopking** [reply]
Great! Thanks for letting me know @jmikrut
